### PR TITLE
AKCORE-78: Add group.share.session.timeout.ms and group.share.heartbeat.interval.ms

### PR DESCRIPTION
**Pull request by adixitconfluent**
### About
AKCORE-78: Add `group.share.session.timeout.ms` and `group.share.heartbeat.interval.ms` configs for share group

### CODE REVIEW

**Review by AndrewJSchofield — APPROVED** [review]
Thanks for the PR. Looks good to me.

File: core/src/main/scala/kafka/server/KafkaConfig.scala [hunk 1]
```diff
@@ -802,6 +808,12 @@ object KafkaConfig {
   val ShareGroupRecordLockPartitionLimitDoc = "Share-group record lock limit per share-partition."
   val ShareGroupMaxGroupsDoc = "The maximum number of share groups."
   val ShareGroupMaxSizeDoc = "The maximum number of consumers that a single share group can accommodate."
+  val ShareGroupSessionTimeoutMsDoc = "The timeout to detect client failures when using the share group protocol."
+  val ShareGroupMinSessionTimeoutMsDoc = "The minimum allowed session timeout."
+  val ShareGroupMaxSessionTimeoutMsDoc = "The maximum allowed session timeout."
+  val ShareGroupHeartbeatIntervalMsDoc = "The heartbeat interval given to the members of a share group."
+  val ShareGroupMinHeartbeatIntervalMsDoc = "The minimum heartbeat interval."
+  val ShareGroupMaxHeartbeatIntervalMsDoc = "The maximum heartbeat interval."
```
Comment (apoorvmittal10) [inline core/src/main/scala/kafka/server/KafkaConfig.scala:817]: @AndrewJSchofield Should we have the elaborated description, may be need to update the KIP as well? i.e. `The minimum heartbeat interval for registered share group members.`

**Review by apoorvmittal10 — APPROVED** [review]

File: core/src/main/scala/kafka/server/KafkaConfig.scala (hunk 1 above)
Comment (AndrewJSchofield) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:817]: Good point. I reckon "for share group members" would work.

Comment (AndrewJSchofield) [reply core/src/main/scala/kafka/server/KafkaConfig.scala:817]: This file has actually been converted into Java in trunk now (or in the process of merging, anyway). There's some merge work building up out there.

Comment (adixitconfluent) [reply core/src/main/scala/kafka/server/KafkaConfig.scala]: Thanks for your comments @apoorvmittal10 @AndrewJSchofield . I have updated the description for `ShareGroupMinHeartbeatIntervalMsDoc`, `ShareGroupMaxHeartbeatIntervalMsDoc`, `ShareGroupMinSessionTimeoutMsDoc` and `ShareGroupMaxSessionTimeoutMsDoc`.

**Pull request merged**